packages/client/hmi-client/src/services/concept.ts

```diff
@@ -154,64 +154,80 @@ const autoEntityMapping = async (
 	// Take out any duplicates
 	const distinctSourceGroundings = [...new Set(allSourceGroundings)];
 	const distinctTargetGroundings = [...new Set(allTargetGroundings)];
-	const allSimilarity = await getEntitySimilarity(
-		distinctSourceGroundings,
-		distinctTargetGroundings
-	);
-	if (!allSimilarity) return result;
-	// Filter out anything with a similarity too small
-	const filteredSimilarity = allSimilarity.filter((ele) => ele.similarity >= acceptableDistance);
-	const validSources: any[] = [];
-	const validTargets: any[] = [];
 
-	// Join results back with source and target
-	filteredSimilarity.forEach((similarity) => {
-		// Find all Sources assosiated with this similarity
+	// Match id strings if there are no groundings in either source or target
+	if (isEmpty(distinctTargetGroundings) || isEmpty(distinctSourceGroundings)) {
```

Comment:
Rather than 
If no grounding use exact mapping
If grounding use grounding

Do you think it would be better to:
Use exact mapping, then whatever is leftover check their groundings?